persistent/persistent.cabal

```diff
@@ -157,7 +157,7 @@ test-suite test
     hs-source-dirs:
         test/
 
-    ghc-options: -Wall
+    ghc-options: -Wall -Wno-incomplete-uni-patterns -dth-dec-file
```

Comment:
What's `-dth-dec-file` do?

Reply:
Ohhh that's super cool! Thanks for the reference 😄 

(full circle: i wrote the bug you diagnosed :joy:)